src/components/SplitVertical/SplitVertical.jsx

```diff
@@ -0,0 +1,392 @@
+ */
+const SplitVertical = createClass({
+	displayName: 'SplitVertical',
+
+	propTypes: {
+		/**
+		 * Appended to the component-specific class names set on the root
+		 * element. Value is run through the `classnames` library.
+		 */
+		className: any,
+		/**
+		 * Direct children must be types {Splitvertical.Leftpane, Splitvertical.Divider, Splitvertical.Leftpane}.
+		 * All content is composed as children of these respective elements.
+		 */
+		children: node,
+		/**
+		 * Render as expanded or collapsed.
+		 */
+		isExpanded: bool,
+		/**
+		 * Allows animated expand and collapse behavior.
+		 */
+		isAnimated: bool,
+		/**
+		 * Called when the user resizes the split with the Divider.
+		 *
+		 * Signature: `(width, { event, props }) => {}`
+		 */
+		onResize: func,
+		/**
+		 * Use this prop to shift the collapsed position by a known value.
+		 */
+		collapseShift: number,
+	},
+
+	components: {
+		/**
+		 * Left pane of the split.
+		 */
+		LeftPane: createClass({
+			displayName: 'SplitVertical.LeftPane',
+			propTypes: {
+				/**
+				 * Any valid React children.
+				 */
+				children: node,
+				/**
+				 * Set width of this pane.
+				 */
+				width: oneOfType([number, string]),
+				/**
+				 * Define this pane as the primary content pane. When the split is collapsed, this pane becomes full width.
+				 */
+				isPrimary: bool,
+			},
+			getDefaultProps(){
+				return {
+					isPrimary: false,
+				};
+			},
+		}),
+
+		/**
+		 * Right pane of the split.
+		 */
+		RightPane: createClass({
+			displayName: 'SplitVertical.RightPane',
+			propTypes: {
+				/**
+				 * Any valid React children.
+				 */
+				children: node,
+				/**
+				 * Set width of this pane.
+				 */
+				width: oneOfType([number, string]),
+				/**
+				 * Define this pane as the primary content pane. When the split is collapsed, this pane becomes full width.
+				 */
+				isPrimary: bool,
+			},
+			getDefaultProps(){
+				return {
+					isPrimary: false,
+				};
+			},
+		}),
+
+		/**
+		 * The area that separates the split panes. Can be dragged to resize them.
+		 */
+		Divider: createClass({
+			displayName: 'SplitVertical.Divider',
+			propTypes: {
+				/**
+				 * Any valid React children.
+				 */
+				children: node,
+			},
+		}),
+	},
+
+	getDefaultProps() {
+		return {
+			isExpanded: true,
+			isAnimated: false,
+			collapseShift: 0,
+			onResize: _.noop,
+		};
+	},
+
+	getInitialState() {
+		return {
+			isAnimated: false,
+		};
+	},
+
+	getPanes() {
+		const { children } = this.props;
+		const {
+			leftPane: leftPaneRef,
+			rightPane: rightPaneRef,
+		} = this.refs;
+
+		const leftPaneElement = _.get(filterTypes(children, SplitVertical.LeftPane), 0, <SplitVertical.LeftPane />)
+		const rightPaneElement = _.get(filterTypes(children, SplitVertical.RightPane), 0, <SplitVertical.RightPane />)
+		let primaryElement, primaryRef;
+		let secondaryElement, secondaryRef;
+
+		if (leftPaneElement.props.isPrimary && !rightPaneElement.props.isPrimary) {
```

Comment:
this looks redundant but it's really just making sure that if the user sets `isPrimary` on both panes (a mistake), it will just do the default behavior, which is to set the right pane as the primary.